static-metric/Cargo.toml

```
@@ -19,7 +19,7 @@ lazy_static = "1.4"
 
 [dev-dependencies]
 criterion = "0.3"
-prometheus = { version = "0.11.0", path = "../" }
+prometheus = { version = "0.12.0", path = "../" }
```

```suggestion
prometheus = { path = "../" }
```

Is the version needed in the first place? In https://github.com/libp2p/rust-libp2p/ we use relative paths only to specify dev-dependencies in the same repository to get around the circular dependency issue.

---

As far as I know it is needed otherwise `cargo` won't let you publish/upload it. But maybe some changed or my knowledge is incomplete.

---

Ah nevermind there is now a special case for dev-dependencies documented at https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#multiple-locations. So yes I think it can be dropped.